### PR TITLE
feat: Add controls to generate OpenAPI definitions from AppMap data

### PR DESCRIPTION
**Pull request by dustinbyrne**
This PR adds both an instructions page and a command to generate OpenAPI definitions for a workspace. Once OpenAPI definitions are generated, a new document is opened containing the definitions. An automated test is present to validate this behavior.

To invoke the new behavior, visit the `Generate OpenAPI definitions` instruction page in the extension sidebar or run `> AppMap: Generate OpenAPI definitions` from the command palette.

### CODE REVIEW

**Comment by petecheslock** [comment]
When using this to generate openapi on the [sample app](https://github.com/land-of-apps/sample_app_6th_ed/tree/codespaces-devcontainer) in the `codespaces-devcontainer` branch the version of the API won't pass the spec because one of the descriptions is not tab/spaced into the correct section of the YAML.  
Screen Shot 2022-07-08 at 11 50 45 AM

The openapi on the left was generated with:
```npx @appland/appmap openapi -o openapi-cli.yml```

the one on the right with openapi button in VS Code.  Both files attached here:

[openapi-from CLI.yml.txt](https://github.com/applandinc/vscode-appland/files/9073414/openapi-from.CLI.yml.txt)
[openapi-from VS Code.yml.txt](https://github.com/applandinc/vscode-appland/files/9073415/openapi-from.VS.Code.yml.txt)

File: src/commands/generateOpenApi.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            return;
+          }
+
+          const binPath = await getBinPath({
+            dependency: ProgramName.Appmap,
+            globalStoragePath: context.globalStorageUri.fsPath,
+          });
+          const openApiCmd = spawn({
+            binPath,
+            args: ['openapi', '-d', workspaceFolder.uri.fsPath],
```
Comment (kgilpin) [inline src/commands/generateOpenApi.ts:56]: Please use `—appmap-dir`since `-d` is going to change to be the working directory argument.

**Comment by petecheslock** [comment]
Strange formatting break when running via the button:
Image on left is via cli - on right is via button

Screen Shot 2022-08-18 at 12 14 38 PM

[openapi.zip](https://github.com/applandinc/vscode-appland/files/9375627/openapi.zip)

**Comment by dustinbyrne** [comment]
This is [sample_app_6th_ed](https://github.com/learnenough/sample_app_6th_ed) right? For some reason I'm unable to recreate.

**Comment by dustinbyrne** [comment]
Though I was able to find what I suspect caused the issue. Here's another release: [appmap-0.40.0-2-dev.zip](https://github.com/applandinc/vscode-appland/files/9384838/appmap-0.40.0-2-dev.zip)

**Comment by petecheslock** [comment]
Looks good - was able to import into postman - visible in the Openapi VS Code extension.  Looks good!

**Pull request merged**